src/models/release-asset.ts

```diff
@@ -0,0 +1,446 @@
+import type {
+	ODataOptionsWithoutCount,
+	OptionsToResponse,
+} from 'pinejs-client-core';
+import type {
+	InjectedDependenciesParam,
+	InjectedOptionsParam,
+	Release,
+	ReleaseAsset,
+} from '..';
+import { limitedMap, mergePineOptions } from '../util';
+import type { ReleaseRawVersionApplicationPair } from './release';
+import { BalenaError } from 'balena-errors';
+import once from 'lodash/once';
+
+type WriteReleaseAssetParams = Omit<ReleaseAsset['Write'], 'asset'> & {
+	asset: File | string;
+};
+type UploadPart = {
+	url: string;
+	chunkSize: number;
+	partNumber: number;
+};
+
+type CommitData = {
+	uuid: string;
+	providerCommitData: {
+		Parts: Array<{ PartNumber: number; ETag: string }>;
+	};
+};
+
+type ReleaseAssetBeginUpload = {
+	asset: {
+		uuid: string;
+		uploadParts: UploadPart[];
+	};
+};
+
+type UploadProgressInfo = {
+	total: number;
+	received: number;
```

Comment:
@otaviojacobi Should this be `uploaded`/`sent` instead of `received`?
PS: I initially fat fingered `received` in my comment, but edited right away to `uploaded` once I realize that I had pasted the wrong thing. I'm sorry for the confusion.

Reply:
@thgreasi I interpreted as `received` by the server which IMO makes sense.... I don't mind changing